scikit-ci.yml

```diff
@@ -17,8 +17,9 @@ before_install:
   travis:
     osx:
       environment:
+        CPPFLAGS: "-stdlib=libstdc++"
         PATH: $<HOME>/.pyenv/versions/$<PYTHON_VERSION>/bin:$<PATH>
-        SETUP_BDIST_WHEEL_ARGS: --plat-name macosx-10.6-x86_64
+        SETUP_BDIST_WHEEL_ARGS: --plat-name macosx-10.13-x86_64
```

Comment:
I couldn't find a website listing what could be the minimum target deployment for a given based SDK. Considered that `xcode12` Travis image is running `macOS 10.15.7` and bundles  `Xcode 12.0.1`    (see [here](https://docs.travis-ci.com/user/reference/osx#macos-version), targeting `10.13` seems "reasonable". 

If we get complains from users, we can re-assess to build wheel compatible with older system.

Also worth noting that macOS `10.13` is old enough to ensure the wheel will be installable in most of the current system:

![image](https://user-images.githubusercontent.com/219043/107652220-56890000-6c4e-11eb-8b8d-c36a78bab6b7.png)

![image](https://user-images.githubusercontent.com/219043/107652752-e9299f00-6c4e-11eb-8566-71b4f4e351be.png)